AK/NumberFormat.cpp

```diff
@@ -11,7 +11,7 @@
 
 namespace AK {
 
-// FIXME: Remove this hackery once printf() supports floats.
+// NOTE: Using this instead of formatting as a floating point since we do not want the rounding behavior
```

Comment:
We had tests that checked for the behavior at the edges (expecting things like 2gb - 1byte rounding to 1.9gb instead of 2.0gb). That behavior seemed reasonable (and somewhat desirable) to me in context of, say, free disk area.

Reply:
1. I'm not sure that we want this behavior (IMO it's misleading and just plain false)
2. The formatter isn't consistent enough to do a proper scientific rounding

So we are quite blocked for now. Anyway, considering the behavior we want, we should either:
 - Enhance the formatter, explicitly show that we are rounding down, and using the standard formatter
 - Enhance the formatter, use it and update tests.

But, that's definitely outside the scope of this PR, so please let this FIXME.